v8go.cc

```diff
@@ -948,18 +949,23 @@ RtnString ValueToDetailString(ValuePtr ptr) {
     return rtn;
   }
   String::Utf8Value ds(iso, str);
-  rtn.string = CopyString(ds);
+  rtn.data = CopyString(ds);
```

Comment:
There are many places that rely on CopyString returning a `const char*` and supporting the two argument types (there are two CopyStrings) so I expect changing the return type would be a much larger change in this PR. Similarly, changing its signature to include another input (length) would impact those locations. Could this PR remain as a fix for NewValue and String with a separate PR to change the internals of CopyString and fix ValueToDetailString?

Reply:
I think all that is needed to fix ValueToDetailString in this PR is the suggested change to set rtn.length and to use the (const char* s, size_t n) version of the std::string constructor in CopyString(String::Utf8Value& value) so that it actually copies the whole string.

```diff
diff --git a/v8go.cc b/v8go.cc
index 7284c0c..e701196 100644
--- a/v8go.cc
+++ b/v8go.cc
@@ -58,7 +58,7 @@ const char* CopyString(String::Utf8Value& value) {
   if (value.length() == 0) {
     return nullptr;
   }
-  return CopyString(*value);
+  return CopyString(*value, value.length());
 }
 
 static RtnError ExceptionError(TryCatch& try_catch,
```

Reply:
Do you mean:
```diff
@@ -58,7 +58,7 @@ const char* CopyString(String::Utf8Value& value) {
   if (value.length() == 0) {
     return nullptr;
   }
-  return CopyString(*value);
+  return CopyString(std::string(*value, value.length()));
```

Reply:
The diff you suggested produces an error because, as I mentioned, it would change the signature of the other CopyString:
```
# rogchap.com/v8go
v8go.cc:61:10: error: no matching function for call to 'CopyString'
v8go.cc:57:13: note: candidate function not viable: requires single argument 'value', but 2 arguments were provided
v8go.cc:49:13: note: candidate function not viable: requires single argument 'str', but 2 arguments were provided
```

Reply:
Oh right, I did mean to use the std::string constructor explicitly like that